packages/react-dom-bindings/src/server/ReactDOMServerFormatConfig.js

```diff
@@ -2001,65 +2034,172 @@ export function writeEndSegment(
+//       // E.g. because the parent was hydrated. That's fine there's nothing to do
+//       // but we have to make sure that we already deleted the container node.
+//       return;
+//     }
+//     // Find the boundary around the fallback. This is always the previous node.
+//     const suspenseNode = suspenseIdNode.previousSibling;
+
+//     if (contentNode) {
+//       // Clear all the existing children. This is complicated because
+//       // there can be embedded Suspense boundaries in the fallback.
+//       // This is similar to clearSuspenseBoundary in ReactDOMHostConfig.
+//       // TODO: We could avoid this if we never emitted suspense boundaries in fallback trees.
+//       // They never hydrate anyway. However, currently we support incrementally loading the fallback.
+//       const parentInstance = suspenseNode.parentNode;
+//       let node = suspenseNode.nextSibling;
+//       let depth = 0;
+//       do {
+//         if (node && node.nodeType === COMMENT_NODE) {
+//           const data = node.data;
+//           if (data === SUSPENSE_END_DATA) {
+//             if (depth === 0) {
+//               break;
+//             } else {
+//               depth--;
+//             }
+//           } else if (
+//             data === SUSPENSE_START_DATA ||
+//             data === SUSPENSE_PENDING_START_DATA ||
+//             data === SUSPENSE_FALLBACK_START_DATA
+//           ) {
+//             depth++;
+//           }
+//         }
 //
-//   const endOfBoundary = node;
+//         const nextNode = node.nextSibling;
+//         parentInstance.removeChild(node);
+//         node = nextNode;
+//       } while (node);
 //
-//   // Insert all the children from the contentNode between the start and end of suspense boundary.
-//   while (contentNode.firstChild) {
-//     parentInstance.insertBefore(contentNode.firstChild, endOfBoundary);
+//       const endOfBoundary = node;
+//
+//       // Insert all the children from the contentNode between the start and end of suspense boundary.
+//       while (contentNode.firstChild) {
+//         parentInstance.insertBefore(contentNode.firstChild, endOfBoundary);
+//       }
+//
+//       suspenseNode.data = '$';
+//     } else {
+//       suspenseNode.data = '$!';
+//       suspenseIdNode.setAttribute('data-dgst', errorDigest)
+//     }
+//
+//     if (suspenseNode._reactRetry) {
+//       suspenseNode._reactRetry();
+//     }
 //   }
-
-//   suspenseNode.data = SUSPENSE_START_DATA;
-//   if (suspenseNode._reactRetry) {
-//     suspenseNode._reactRetry();
+//   return function completeBoundary(suspenseBoundaryID, contentID, styleResources){
+//     const contentNode = document.getElementById(contentID);
+//     // We'll detach the content node so that regardless of what happens next we don't leave in the tree.
+//     // This might also help by not causing recalcing each time we move a child from here to the target.
+//     contentNode.parentNode.removeChild(contentNode);
```

Comment:
When we remove this we might cause some recalc to figure out how this applies to the tree and might need to reapply some selectors.

It's probably better to do this at the same time we do the rest of the reveal. So this should move into flipBoundary.